plugin/NannouLaserDevice.h

```diff
@@ -0,0 +1,294 @@
+/* -------------------------------------- . ---------------------------------- .
+| Filename : NannouLaserDevice.h          |                                    |
+| Author   : MindBuffer                   |                                    |
+| Started  : 17/04/2020 13:05             |                                    |
+` --------------------------------------- . --------------------------------- */
+
+#ifndef SMODE_NANNOU_LASER_DEVICE_H_
+# define SMODE_NANNOU_LASER_DEVICE_H_
+
+#include "LaserDevice.h"
+#include "../smode_laser/smode_laser.h"
+
+namespace smode
+{
+class NannouLaserDevice : public LaserDevice
+{
+public:
+  NannouLaserDevice(const DeviceIdentifier& identifier, laser::Api* _api, laser::DetectedDac _dac)
+    : LaserDevice(identifier), dacPointsPerSecond(10000), latencyPoints(166), targetFps(60), blankDelayPoints(10), distancePerPoint(0.1), anglePerPoint(0.6), api(_api), dac(_dac), callback_data(std::make_shared<CallbackData>())
+  {
+    dacPointsPerSecond.setParent(this);
+    latencyPoints.setParent(this);
+    targetFps.setParent(this);
+    distancePerPoint.setParent(this);
+    blankDelayPoints.setParent(this);
+    anglePerPoint.setParent(this);
+  }
+  NannouLaserDevice() {}
+
+  bool hasVariableConstraints() const override
+    {return true;}
```

Comment:
not necessary as long as there is no interaction between at least two different member variables.
(so you should be able to just remove those two lines in your case)